src/dali/plugin/input/rest/binance_com.py

```diff
@@ -0,0 +1,1295 @@
+            fiat_out_no_fee: RP2Decimal = RP2Decimal(str(transaction[_COST]))
+            fiat_fee: RP2Decimal = RP2Decimal(crypto_fee)
+            spot_price: RP2Decimal = RP2Decimal(str(transaction[_PRICE]))
+
+            out_transaction_list.append(
+                OutTransaction(
+                    plugin=self.__BINANCE_COM,
+                    unique_id=transaction[_ID],
+                    raw_data=json.dumps(transaction),
+                    timestamp=self._rp2timestamp_from_ms_epoch(transaction[_TIMESTAMP]),
+                    asset=out_asset,
+                    exchange=self.__BINANCE_COM,
+                    holder=self.account_holder,
+                    transaction_type=Keyword.SELL.value,
+                    spot_price=str(spot_price),
+                    crypto_out_no_fee=str(crypto_out_no_fee),
+                    crypto_fee=str(crypto_fee),
+                    crypto_out_with_fee=str(crypto_out_with_fee),
+                    fiat_out_no_fee=str(fiat_out_no_fee),
+                    fiat_fee=str(fiat_fee),
+                    fiat_ticker=trade.quote_asset,
+                    notes=(f"{notes + ';' if notes else ''} Fiat sell of {trade.base_asset} with {trade.quote_asset}."),
+                )
+            )
+
+        else:
+            # Binance does not report the value of transaction in fiat
+            out_transaction_list.append(
+                OutTransaction(
+                    plugin=self.__BINANCE_COM,
+                    unique_id=transaction[_ID],
+                    raw_data=json.dumps(transaction),
+                    timestamp=self._rp2timestamp_from_ms_epoch(transaction[_TIMESTAMP]),
+                    asset=out_asset,
+                    exchange=self.__BINANCE_COM,
+                    holder=self.account_holder,
+                    transaction_type=Keyword.SELL.value,
+                    spot_price=Keyword.UNKNOWN.value,
+                    crypto_out_no_fee=str(crypto_out_no_fee),
+                    crypto_fee=str(crypto_fee),
+                    crypto_out_with_fee=str(crypto_out_with_fee),
+                    fiat_out_no_fee=None,
+                    fiat_fee=None,
+                    notes=(
+                        f"{notes + '; ' if notes else ''} Sell side of conversion from "
+                        f"{conversion_info}"
+                        f"({in_asset} in-transaction unique id: {transaction[_ID]}"
+                    ),
+                )
+            )
+
+    def _process_transfer(self, transaction: Any, intra_transaction_list: List[IntraTransaction]) -> None:
+
+        # This is a CCXT list must convert to string from float
+        amount: RP2Decimal = RP2Decimal(str(transaction[_AMOUNT]))
+
+        if transaction[_TYPE] == _DEPOSIT:
+            intra_transaction_list.append(
+                IntraTransaction(
+                    plugin=self.__BINANCE_COM,
+                    unique_id=transaction[_TXID],

```

Comment:
The value assigned to `unique_id` here is the transaction hash, correct? I know we discussed this in the past but I wanted to double check we're on the same page. The `unique_id` how-to topic (which is the context for this discussion) is so important that I recently decided to formalize it into a FAQ: https://github.com/eprbell/dali-rp2/blob/main/docs/developer_faq.md#how-to-fill-the-unique-id-field. The same applies to all other unresolved transactions (i.e. transactions that have UNKNOWN from/to fields).

Reply:
Yes, `_TX_ID` (its new name after adding an underscore) is the transaction hash. The Binance internal id is `_ID` if there is one.
The transaction resolver has matched this with external transaction hashes I have for mining data, so I can validate that this works as expected.

Reply:
Sounds good.